web_poet/testing/pytest.py

```diff
@@ -144,6 +153,38 @@ def reportinfo(self):
         )
 
 
+class WebPoetExpectedException(_WebPoetItem):
+    def runtest(self) -> None:
+        self.fixture.assert_toitem_exception()
+
+    def reportinfo(self):
+        return (
+            self._path,
+            0,
+            f"{self.fixture.short_name}: to_item raises {self.fixture.get_expected_exception().__class__.__name__}",
+        )
+
+    def repr_failure(self, excinfo, style=None):
+        expected = self.fixture.get_expected_exception()
+        if isinstance(excinfo.value, ExceptionNotRaised):
+            return (
+                f"to_item() didn't raise an exception."
+                f" {get_fq_class_name(type(expected))} was expected."
+            )
+        if isinstance(excinfo.value, WrongExceptionRaised):
+            got = excinfo.value.__cause__
+            inner_excinfo = pytest.ExceptionInfo.from_exc_info(
+                (type(got), got, got.__traceback__)
+            )
+            return (
+                f"to_item() raised a wrong exception. Expected"
+                f" {get_fq_class_name(type(expected))}, got"
+                f" {get_fq_class_name(type(got))}.\n\n"
+                + str(super().repr_failure(inner_excinfo, style))
```

Comment:
pytest code doesn't allow us to make a report in the way I wanted so here is a custom "header" and the traceback from pytest converted to a string (which I think only removes colors and an exception name after "test-1::TO_ITEM_RAISES" in the "short test summary info"). I think formatting the traceback manually doesn't bring us anything but loses the ability to customize it via `--tb=`.